src/utl/include/utl/LoggerCommands.h

```diff
@@ -0,0 +1,55 @@
+/////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (c) 2022, The Regents of the University of California
+// All rights reserved.
+//
+// BSD 3-Clause License
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+// This is only for including into Logger.h and Logger-py.i
```

Comment:
I suppose you meant Logger.i not Logger.h.  In that case just put them in a common .i file that they both include.   There is no need to have them be a API to utl in C++.

Reply:
Not certain exactly what you are asking here. If the C++ code is put into an %inline %{ %} block in a common .i file and then included in both .i files, I believe that they will be multiply defined. Or did you mean to to have just the %include "Logger.h" and the %include "LoggerCommands.h" in the common .i file?

Reply:
We already do this in odb, for example polygon.i  is in tcl & python.  The swig wrappers have different signatures, eg

SWIGINTERN PyObject *_wrap_shrinkSet(PyObject *self, PyObject *args) {
SWIGINTERN int _wrap_shrinkSet(ClientData clientData SWIGUNUSED, Tcl_Interp *interp, int objc, Tcl_Obj *CONST objv[]) {

Reply:
But polygon.i includes a swig_common.h with a corresponding swig_common.cpp to avoid the multiple definitions, I believe. This is the same as a LoggerCommands.h / LoggerCommands.cpp, which we could rename to LoggerCommon.h and LoggerCommon.cpp if that's better.

Reply:
You are right so let's just leave it in c++.  However the header doesn't need to be in the public include area and can just go in src/

Reply:
OK. Let me do that.